internal/api/verification.go

```diff
@@ -0,0 +1,77 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/iden3/go-iden3-core/v2/w3c"
+
+	"github.com/polygonid/sh-id-platform/internal/log"
+)
+
+// CheckVerification returns CheckVerificationResponse or Provide Query
+func (s *Server) CheckVerification(ctx context.Context, request CheckVerificationRequestObject) (CheckVerificationResponseObject, error) {
+	//validation
+	issuerDID, err := w3c.ParseDID(request.Identifier)
+	if err != nil {
+		log.Error(ctx, "parsing issuer did", "err", err, "did", request.Identifier)
+		return CheckVerification400JSONResponse{N400JSONResponse{Message: "invalid issuer did"}}, nil
+	}
+
+	verificationQueryID, err := uuid.Parse(request.Params.Id.String())
+	if err != nil {
+		log.Error(ctx, "invalid verification query ID", "err", err)
+		return CheckVerification400JSONResponse{N400JSONResponse{Message: "invalid verification query ID"}}, nil
+	}
+
+	// Use the VerificationService to check if there's an existing response or get the query
+	response, query, err := s.verificationService.CheckVerification(ctx, *issuerDID, verificationQueryID)
+	if err != nil {
+		log.Error(ctx, "checking verification", "err", err)
+		return CheckVerification500JSONResponse{N500JSONResponse{Message: err.Error()}}, nil
+	}
+
+	if response != nil {
+		return CheckVerification200JSONResponse{union: json.RawMessage(fmt.Sprintf(`{"status": "validated", "response": %v}`, response))}, nil
+	}
+
+	if query != nil {
+		return CheckVerification200JSONResponse{union: json.RawMessage(fmt.Sprintf(`{"query": %v}`, query))}, nil
+	}
+
+	return CheckVerification404JSONResponse{N404JSONResponse{Message: "Verification query not found"}}, nil
```

Comment:
default response should be 500. To return error 404 we'd check the error from the repository /service layer